sc-memory/sc-memory/sc_link.hpp

```diff
@@ -96,15 +96,20 @@ class ScLink
   {
     if (!IsType<Type>())
     {
-      SC_THROW_EXCEPTION(utils::ExceptionInvalidParams,
-                         "You use incorrect type. Use IsType<>() to check it");
+      SC_THROW_EXCEPTION(utils::ExceptionInvalidType,
+                         "You've used incorrect type. Use IsType<>() to check it");
     }
     ScStreamPtr const stream = m_ctx.GetLinkContent(m_addr);
+
+    // Check for empty content.
+    if (!stream)
+      return Type();
+
     Type result;
-    if (!stream || !Stream2Value(stream, result))
+    if (!Stream2Value(stream, result))
     {
       SC_THROW_EXCEPTION(utils::ExceptionCritical,
-                         "Error to get value of " + std::to_string(m_addr.Hash()));
+                         "Failed to convert the value of " + std::to_string(m_addr.Hash()));
```

Comment:
Failed to convert? What is a source and target types?
Providing of error should not expose implementation. I think better to return just two types of error:
- Invalid type. We request `type1` but in knowledge base content marked with `type2`
- An common error: we can't get content of sc-link.

Thats all. In other case this error will be changes with changing of internal implementation. This is not very good